libkbfs/data_types.go

```diff
@@ -195,6 +195,9 @@ const (
 	// FilesWithHolesDataVer is the data version for files
 	// with holes.
 	FilesWithHolesDataVer DataVer = 2
+	// BigFilesDataVer is the data version for files that have
+	// multiple levels of indirection.
+	BigFilesDataVer DataVer = 3
```

Comment:
> b) Indirect blocks of depth 2 (meaning one indirect block pointing to all direct blocks) can be v1 (if it has no holes), v2 (if it has holes), or v3 (if it has at least one indirect ptr with TargetBlockType = Direct [although if it has one, all of them should be]). However, all its indirect pointers will have DataVer 1, by a).

A "depth 2" indirect block should never be v3, since older clients can read it just fine.  If there is one direct block marked as such, the block will be either v1 or v2 depending on the holes situation.

Otherwise this seems accurate, and I'll add it to the file as a comment, thanks!